articles/introduction-to-c#/index .md

```diff
@@ -0,0 +1,231 @@
+ ```C#
+ namespace HelloWorld{}
+ ``` 
+
+Namespaces are used in C# to seperate and organize code. They are containers consisting of methods, classes and other namespaces.
+
+**Line 4:**
+ ``` C#
+ class Program{}
+ ```
+
+A class is a container that contains methods and data. Since C# is Object-oriented programming, creating a class is mandatory for each program. In our case, we are creating a class called `Program`.
+
+**Line 5:** 
+```C#
+static void Main(string[] args)
+```
+Execution of programs begins from the `Main` method. Every C# program must have the `Main` method.
+
+**Line 6:**
+ ```C#
+ Console.WriteLine("Hello World!");
+ ```
+
+`Console.WriteLine()` is used to print out text on the console.It prints a string and moves to the next line. In our case, it is used to output *"Hello World"*.
+
+We can also use `Console.Write()` that print a string in the same line without moving to the next line.  You can learn more on Console.WriteLine() and Console.Write() [here](https://www.programiz.com/csharp-programming/basic-input-output).
+
+**Line 7** 
+```C#
+Console.ReadLine();
+```
+
+`Console.ReadLine()` is used to read the next characters in the input system.
+In the above program, as soon as the user enters a character(s) and presses the ENTER key, it terminates.
+
+`Console.Read()`can also be used to read the next character in the input system.
+
+```Console.ReadKey()``` can also be used to obtain the next key entered by the user. It is mostly used to hold the console for the user until he presses a key.
+
+**Note:**
+- Each line in C# ends with a semi-colon(;).
+- C# follows a top to bottom order when executing the program.
+- C# is case-sensitive as it treats capitalized and lowercase characters in a different way.e.g `program` and `Program` are different things.
+
+**Congratulations!** You just wrote your first C# program.
+
+Let's write another program that adds two numbers entered by the user.
+```C#
+//This is a C# program that adds two numbers
+using System;
+
+    namespace HelloWorld
+        {
+            class Program
+            {       
+                static void Main(string[] args){
+                    Console.WriteLine("Enter your first number: ");
+                    int num1= Convert.ToInt32(Console.ReadLine());
+                    Console.WriteLine("Enter your second number: ");
+                    int num2= Convert.ToInt32(Console.ReadLine());
+                    Console.WriteLine("Your Sum is "num1 + num2);//prints the sum of the two numbers 
+                    Console.ReadLine();
+               }
+         }
+    }
+
+```
+
+
+Here's how your program output will look like:
+
+```
+Enter your first number: 2
+
+Enter your second number: 5
+
+Your Sum Is 7
+```
+Let us look at the new things in our program:
+
+```c#
+int num1= Convert.ToInt32(Console.ReadLine());
+```
+
+`int num1` means that we are creating a memory location to hold an integer (int). In our case, we have named the integer **num1**.
```

Comment:
We are not naming the Integer, we are naming the memory location that holds the integer. Make sure to rewrite this.